Editor/UIToolkit/PreferencesWindow/PackagePreferencesWindow.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            m_TabController.AddTab(label, content);
+        }
+
+        /// <summary>
+        /// Activate tab by name
+        /// </summary>
+        /// <param name="name">Early specified tab name</param>
+        protected void ActivateTab(string name)
+        {
+            m_TabController.ActivateTab(name);
+        }
+
+        /// <summary>
+        /// Set the flexible growth property of tabs content container
+        /// </summary>
+        /// <param name="styleFloat"></param>
+        protected void ContentContainerFlexGrow(StyleFloat styleFloat)
+        {
+            m_TabController.ContentContainerFlexGrow(styleFloat);
+        }
+
+        /// <summary>
+        /// Overrides SettingsProvider.OnActivate.
+        /// </summary>
+        protected abstract void OnActivate(string searchContext, VisualElement rootElement);
+
+        /// <summary>
+        /// Overrides SettingsProvider.OnDeactivate.
+        /// </summary>
+        protected abstract void OnDeactivate();
+        
+        void OnActivateWindow(string searchContext, VisualElement rootElement)
+        {
+            UIToolkitEditorUtility.CloneTreeAndApplyStyle(rootElement,
+                $"{PluginsDevKitPackage.UIToolkitPath}/SettingsWindow/PackageSettingsWindow");
+
+            // Hide search bar from PackageSettingsWindow. In preferences we already have search bar
+            // and it's value in "searchContext" parameter
+            var searchBar = rootElement.Q<ToolbarSearchField>();
+            if (searchBar != null)
+            {
+                searchBar.style.visibility = Visibility.Hidden;
+            }
+            
+            var packageInfo = GetPackageInfo();
+            rootElement.Q<Label>("display-name").text = packageInfo.displayName;
+            rootElement.Q<Label>("description").text = packageInfo.description;
+            rootElement.Q<Label>("version").text = $"Version: {packageInfo.version}";
+
+            m_TabController = new TabController(rootElement);
+        }
+
+        void OnActivateHandler(string searchContext, VisualElement rootElement)
+        {
+            OnActivate(searchContext, rootElement);
+
+            EditorApplication.delayCall += () =>
+            {
+                m_TabController.RefreshActiveTab();
+            };
+        }
+
+        SettingsProvider ConstructSettingsProvider()
+        {
+            var packageInfo = GetPackageInfo();
+            var settingsProvider = new SettingsProvider(SettingsPath, Scope, packageInfo.keywords)
+            {
+                label = packageInfo.displayName,
+            };
+
+            settingsProvider.activateHandler += OnActivateWindow;
+            settingsProvider.activateHandler += OnActivateHandler;
+            settingsProvider.deactivateHandler += OnDeactivate;
+
+            return settingsProvider;
+        }
+
+        [SettingsProviderGroup]
+        static SettingsProvider[] RegisterSettingsProviderGroup()
```

Comment:
The only think I would do is to move this to a spearate file.
but that's more like a suggestion / perosnal preference.

Reply:
Not sure, the idea is to keep it inside PackagePreferencesWindow because we have everything in one place. But I agree that this class might grow at some point without decomposition.
I would be glad if we keep it this way for now as a simplest solution and see later how to deal with that.